src/braket/experimental/algorithms/quantum_partition_function/quantum_partition_function.py

```diff
@@ -0,0 +1,202 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import math
+from typing import Any, Dict, List
+
+import networkx as nx
+from braket.circuits import Circuit, circuit
+
+
+@circuit.subroutine(register=True)
+def quantum_partition_function(
+    function: str,
+    qubits: List[int],
+) -> Circuit:
+    """Creates two kinds of circuits for quantum partition function:
+      1) The shor's algorithm for irreducible cyclic cocycle code (ICCC)-check
+      2) The quantum fourier transformation circuit for checking gamma values
+
+    Args:
+        function (str): Define the function of this circuit, it can be 'shor' or 'qft'
+        qubits (List[int]) : Qubits defining the circuit
+
+    Returns:
+        Circuit: Circuit object that implements the Quantum Parition Estimation algorithm
+    """
+
+    circuit = None
+
+    if function == "shor":
+        print("shor's algorithm for iccc-check hasn't been implemented!")
+    elif function == "qft":
+        print("qft circuit for checking gamma values")
+        circuit = _quantum_fourier_transform(qubits)
+    else:
+        raise ValueError(
+            f"Wrong function value {function}, can only be one \
+                        of shor and qft"
+        )
+
+    return circuit
+
+
+@circuit.subroutine(register=True)
+def _quantum_fourier_transform(
+    qft_qubits: List[int],
+) -> Circuit:
+    """
+    Construct a circuit object corresponding to the Quantum Fourier Transform (QFT)
+    algorithm, applied to the argument qubits.
+
+    Args:
+        qft_qubits (List[int]): Qubits on which to apply the QFT
+
+    Returns:
+        Circuit: Circuit object that implements the Quantum Fourier Transform (QFT)
+    """
+    qftcirc = Circuit()
+
+    # get number of qubits
+    num_qubits = len(qft_qubits)
+
+    for k in range(num_qubits):
+        # First add a Hadamard gate
+        qftcirc.h(qft_qubits[k])
+
+        for j in range(1, num_qubits - k):
+            angle = 2 * math.pi / (2 ** (j + 1))
+            qftcirc.cphaseshift(qft_qubits[k + j], qft_qubits[k], angle)
+
+    # Then add SWAP gates to reverse the order of the qubits:
+    for i in range(math.floor(num_qubits / 2)):
+        qftcirc.swap(qft_qubits[i], qft_qubits[-i - 1])
+
+    return qftcirc
+
+
+def run_quantum_partition_function(
+    potts_model: dict,
```

Comment:
Maybe you should consider defining ``potts_model`` as an object instead. Otherwise it is hard for a user to tell what should go in here